src/features/auth/register-form/use-register-form-state.ts

```diff
@@ -0,0 +1,36 @@
+import { useState, ChangeEvent } from "react";
+
+export default function useRegisterFormState() {
+    const [name, setName] = useState<string>("");
+    const [email, setEmail] = useState<string>("");
+    const [password, setPassword] = useState<string>("");
+    const [confirmPassword, setConfirmPassword] = useState<string>("");
+
+    const onChangeName = (event: ChangeEvent<HTMLInputElement>) => {
+        setName(event.target.value);
+    }
+
+    const onChangeEmail = (event: ChangeEvent<HTMLInputElement>) => {
+        setEmail(event.target.value);
+    };
+
+    const onChangePassword = (event: ChangeEvent<HTMLInputElement>) => {
+        setPassword(event.target.value);
+    };
+
+    const onChangeConfirmPassword = (event: ChangeEvent<HTMLInputElement>) => {
+        setConfirmPassword(event.target.value);
+    }
```

Comment:
Hmmm, no estoy seguro, pero quiza deberian ser callbacks? por el tema de usarlos dentro de un `useEffect`

```suggestion
    const onChangeName = useCallback((event: ChangeEvent<HTMLInputElement>) => {
        setName(event.target.value);
    }, []);

    const onChangeEmail = useCallback((event: ChangeEvent<HTMLInputElement>) => {
        setEmail(event.target.value);
    }, []);

    const onChangePassword = useCallback((event: ChangeEvent<HTMLInputElement>) => {
        setPassword(event.target.value);
    }, []);

    const onChangeConfirmPassword = useCallback((event: ChangeEvent<HTMLInputElement>) => {
        setConfirmPassword(event.target.value);
    }, []);
```